File: Ilya-Dudnikov/term_2/block_2/task_1/lib/src/main/java/deck/Deck.java
package deck;

import card.Card;
import card.Suits;

import java.util.ArrayList;
import java.util.Collections;

public class Deck implements IDeck {
	private ArrayList<Card> cardsArray;

	public Deck() {
		cardsArray = new ArrayList<>();
		for (var cardSuit : Suits.values()) {
			for (int rank = 2; rank <= 14; rank++) {
				cardsArray.add(new Card(rank, cardSuit));
			}
		}
	}

	public Card getCardAt(int pos) {
		return cardsArray.get(pos);
	}

	public void shuffle() {
		Collections.shuffle(cardsArray);
	}
}

File: IvanShurenkov/Second Term/Block3/Task1/lib/base/src/main/kotlin/lib/blackjack/base/Deck.kt
package lib.blackjack.base

import lib.blackjack.base.state.CardRank
import lib.blackjack.base.state.CardSuit

class Deck(cntDecks: UInt) {
    private var deck: Array<Card> = emptyArray()
    private var pos: Int = -1

    init {
        for (i in 1..cntDecks.toInt()) {
            for (suit in CardSuit.values()) {
                for (rank in CardRank.values()) {
                    deck += Card(rank, suit)
                }
            }
        }
    }

    fun getCard(): Card {
        pos++
        return deck[pos]
    }

    fun remain(): Int {
        return deck.size - (pos + 1)
    }

    fun shaffle() {
        pos = -1
        deck.shuffle()
    }
}
